python/grass/jupyter/timeseriesmap.py

```diff
@@ -281,46 +282,59 @@ def _render_layer(self, layer, filename):
         if self._legend:
             self._render_legend(img)
 
-    def render(self):
-        """Renders image for each time-step in space-time dataset.
+    def _render_layer_worker(self, date, layer, filename):
+        """Function to render a single layer."""
+        shutil.copyfile(self.base_file, filename)
+        if layer == "None":
+            self._render_blank_layer(filename)
+        else:
+            self._render_layer(layer, filename)
+        return date, filename
 
-        Save PNGs to temporary directory. Must be run before creating a visualization
-        (i.e. show or save). Can be time-consuming to run with large
-        space-time datasets.
-        """
+    def _get_num_cores(self):
+        """Get the number of available cores."""
+        try:
+            num_cores = len(os.sched_getaffinity(0))
+        except AttributeError:
+            num_cores = multiprocessing.cpu_count()
+        return num_cores
```

Comment:
```suggestion
        try:
            return len(os.sched_getaffinity(0))
        except AttributeError:
            return multiprocessing.cpu_count()
```

Reply:
You will need this method for seriesmap too, so this should go to utils, this will also fix CI style checking:
```
grass/jupyter/timeseriesmap.py:294:4: R0201: Method could be a function (no-self-use)
```